src/dotnet/APIView/APIViewWeb/Extensions/LinqExtensions.cs

```diff
@@ -0,0 +1,54 @@
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using APIViewWeb.Helpers;
+using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace APIViewWeb.Extensions
+{
+    public static class LinqExtensions
+    {
+        public static IEnumerable<T> InterleavedUnion<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            var firstList = first.ToList();
+            var secondList = second.ToList();
+            var result = new List<T>();
+
+            int i = 0, j = 0;
+            while (i < firstList.Count && j < secondList.Count)
+            {
+                if (firstList[i].Equals(secondList[j]))
+                {
+                    result.Add(firstList[i]);
+                    i++;
+                    j++;
+                }
+                else if (secondList.Contains(firstList[i]))
+                {
+                    result.Add(secondList[j]);
```

Comment:
I can't figure out what this contains check is doing here, followed by adding the `j` index? Also I don't exactly understand the usage context or if time complexity is a concern with the contains check per element.

Reply:
AM trying to create a Union of the two sets without losing the order of occurrence. I can perhaps improve in this later.